var/spack/repos/builtin/packages/glvis/package.py

```diff
@@ -0,0 +1,138 @@
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class Glvis(Package):
+    """GLVis: an OpenGL tool for visualization of FEM meshes and functions"""
+
+    homepage = "http://glvis.org/"
+    url      = "https://github.com/GLvis/glvis/archive/v3.3.tar.gz"
+
+    # glvis (like mfem) is downloaded from a URL shortener at request
+    # of upstream author Tzanio Kolev <tzanio@llnl.gov>.  See here:
+    # https://github.com/mfem/mfem/issues/53
+    #
+    # The following procedure should be used to verify security when a
+    # new version is added:
+    #
+    # 1. Verify that no checksums on old versions have changed.
+    #
+    # 2. Verify that the shortened URL for the new version is listed at:
+    #    http://glvis.org/download/
+    #
+    # 3. Use http://getlinkinfo.com or similar to verify that the
+    #    underling download link for the latest version comes has the
+    #    prefix: http://glvis.github.io/releases
+    #
+    # If this quick verification procedure fails, additional discussion
+    # will be required to verify the new version.
+
+    version('3.3',
+            '1201a76d1b0c38240186c06f6478de77',
+            url='http://goo.gl/C0Oadw',
+            extension='.tar.gz')
+
+    version('3.2',
+            '4575a03a50d4730c07b5d4018e936707',
+            url='http://goo.gl/hzupg1',
+            extension='.tar.gz')
+
+    version('3.1',
+            '3ce8a53209c6593e066e568dbbf2bbf6',
+            url='http://goo.gl/gQZuu9',
+            extension='tar.gz')
+
+    variant('screenshots',
+            default='png',
+            values=('none', 'png', 'tiff'),
+            description='Backend used for screenshots (none = disabled)')
+    variant('fonts', default=True,
+            description='Build with font support via freetype & fontconfig')
+
+    depends_on('mfem@3.1', when='@3.1')
+    depends_on('mfem@3.2', when='@3.2')
+    depends_on('mfem@3.3', when='@3.3')
+
+    depends_on('gl')
+    depends_on('glu')
+    depends_on('libx11')
+
+    depends_on('libpng', when='screenshots=png')
+    depends_on('libtiff', when='screenshots=tiff')
+    depends_on('freetype', when='+fonts')
+    depends_on('fontconfig', when='+fonts')
+
+    def install(self, spec, prefix):
+
+        def yes_no(s):
+            return 'YES' if self.spec.satisfies(s) else 'NO'
+
+        mfem_prefix = self.spec['mfem'].prefix
+        config_mk_prefix = mfem_prefix
+        if spec.satisfies('^mfem@3.3.2', strict=True):
```

Comment:
This line should really be `if spec.satisfies('^mfem@3.3.2:', strict=True):`, but apparently `mfem@3.3` satisfies this condition, which seems like a bug.

Reply:
It may be easier to check directly if `config.mk` is in `mfem_prefix`, and if not, then use `mfem_prefix+'/share/mfem'`.